### DISCUSSION TITLE
Scale object X-axis

### DISCUSSION THREAD

**Discussion by KenyC** [reanimate/reanimate#205] [Q&A] [answered]
Hi,

Thanks for the wonderful package! I was trying to follow the equation tutorial and trying to make an animation where an ``Object`` is scaled down on its X-axis only. There does not seem to be an ``oScaleX`` corresponding to ``oScale``. I managed to get my way with the following:

```haskell
oTweenS objectToScale 1 $ \t -> do
           oContext %= \origin -> (scaleXY (1.0 + t) 1.0) . origin
```

But that does not seem very efficient ; is there a more intelligent way to do that?

**Comment by lemmih** [comment, marked as answer]
Yeah, there's isn't a built-in function for that yet. The whole object/sprite system is being rewamped, though, and will have more consistent ways of being manipulated in the future.
